### PR TITLE
fix: Move v1 conversion prompt into initialization

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Prompting the user to convert v1 profiles during `readConfigFromDisk` can cause the prompt to appear in scenarios where it shouldn't. One example is when making changes to a project config in one Zowe Explorer window, while another window is opened with Zowe Explorer & no team configs.

We should only prompt the user to convert their v1 profiles once during initialization. This PR moves the convert prompt logic into the `ProfilesUtils.handleV1MigrationStatus` function.

## Release Notes

Milestone: 3.0.2

Changelog:

- Fixed an issue where Zowe Explorer sometimes prompted the user to convert v1 profiles when changes were made to a Team Config after initialization. [#3246](https://github.com/zowe/zowe-explorer-vscode/pull/3246)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by JillieBeanSim — DISMISSED** [review]
LGTM! thanks for fix @traeok

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ### Bug fixes
 
+- Fixed an issue where Zowe Explorer sometimes prompted the user to convert v1 profiles when changes were made to a Team Config after initialization. [#3246](https://github.com/zowe/zowe-explorer-vscode/pull/3246)
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md:13]: V1 should be capitalized. "Team Config" should be "team configuration."

**Review by anaxceron — APPROVED** [review]
Suggesting minor edits to the changelog. Thank you @traeok

**Review by JillieBeanSim — APPROVED** [review]

**Review by t1m0thyj — COMMENTED** [review]
Seeing an issue when testing migration in a VM, not sure if this behavior is expected:
* Have environment with V1 profiles
* Install Zowe Explorer V1 and verify that profile shows in trees
* Install Zowe Explorer V3 (built from this branch)
* In VS Code notification, select "Restart Extensions"
* In ZE modal dialog, select "Create New"
* Notice that new config is created but DS and Jobs trees failed to load

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok! Just needed to reset VS Code in my VM since it had settings left over from a previous V3 migration test 😋

**Comment by JillieBeanSim** [comment]
Would the VM scenario be considered an edge case or is this a regression in that environment from 3.0 or 3.0.1 @t1m0thyj ? 
If its a regression we should solve in this PR
If edge case possibly could be solved as separate issue
